bot.js

```diff
@@ -239,16 +239,34 @@ spawnPokemon = async (message, designatedChannel) => {
                     .setThumbnail(collected.first().author.avatarURL())
                     .setImage(generatedPKMNData.sprite);
                 let out = "```" + generateSpaces("LV." + generatedPKMNData.level, 6) + "| IV:" + generateSpaces(generateIVSummary(generatedPKMNData.ivs), 5) + "| " + generatedPKMNData.nature[0] + "```";
-                embed.addField("Info:", out);
+                embed.addField("Info:" + '\n' + "PokeDollars Obtained: " + parseInt(generatedPKMNData.level * 2),out, out);
+              //  embed.addField("PokeDollars Obtained:" + parseInt(generatedPKMNData.level * 2),out)
                 channelRef.send({embed});
 
+                /**
+                 * Handle Coins
+                 */
+
+                const snapshot = await db.collection('users').doc(collected.first().author.id).get().then((doc) => {
+                    return {id: doc.id, ...doc.data()}
+                })
+
+                let dollarSnap;
+
+                if(isNaN(snapshot.pDollar)){
```

Comment:
Would be better to check if the user's query data has the pDollar key, and if it doesnt have the key, update the user's document to have it. Here's an example of checking an object has a specific key: https://stackoverflow.com/questions/455338/how-do-i-check-if-an-object-has-a-key-in-javascript

As long as the pDollar property of a user is initialized to be a number, and a string is never appended to it, there is no need to check if it's a number.